storage/bootstrap/bootstrapper/commitlog/source.go

```diff
@@ -256,3 +260,31 @@ func (s *commitLogSource) Read(
 
 	return bootstrapResult, nil
 }
+
+func newReadCommitLogPredicate(
+	ns namespace.Metadata,
+	shardsTimeRanges result.ShardTimeRanges,
+	opts Options,
+) commitlog.ReadCommitLogPredicate {
+	return func(commitLogTime time.Time) bool {
+		// Minimum and maximum times for which we want to bootstrap
+		shardMin, shardMax := shardsTimeRanges.MinMax()
+
+		// How far into the past or future a commitlog might contain a write for a
+		// previous or future block
+		bufferPast := ns.Options().RetentionOptions().BufferPast()
+		bufferFuture := ns.Options().RetentionOptions().BufferFuture()
+
+		blockSize := opts.CommitLogOptions().BlockSize()
```

Comment:
@prateek This is correct right? I care about the blockSize of the commitlog, not of the retention?